src/QLPlan.actor.cpp

```diff
@@ -699,22 +699,23 @@ ACTOR static Future<Void> doNonIsolatedRO(PlanCheckpoint* outerCheckpoint,
 			state FutureStream<Reference<ScanReturnedContext>> docs = subPlan->execute(innerCheckpoint.getPtr(), dtr);
 			state FlowLock* innerLock = innerCheckpoint->getDocumentFinishedLock();
 			state bool first = true;
-			state Future<Void> timeout = delay(3.0, TaskMaxPriority);
+			state Future<Void> timeout = delay(3.0, g_network->getCurrentTask() + 1);
```

Comment:
`g_network->getCurrentTask() + 1` could still give you a task priority lower than `TaskDefaultOnMainThread` right?

Reply:
Idea is current task is running at `TaskDefaultMainThread` as that’s the one setting the promise stream.